hpx/runtime_distributed.hpp

```diff
@@ -346,45 +393,98 @@ namespace hpx
         notification_policy_type get_notification_policy(
             char const* prefix) override;
 
+        std::uint32_t get_locality_id(error_code& ec) const override
+        {
+            return agas::get_locality_id(ec);
+        }
+
+        std::size_t get_num_worker_threads() const override
+        {
+            error_code ec(lightweight);
+            return static_cast<std::size_t>(
+                agas_client_.get_num_overall_threads(ec));
+        }
+
+        std::uint32_t get_num_localities(
+            hpx::launch::sync_policy, error_code& ec) const override
+        {
+            return agas_client_.get_num_localities(ec);
+        }
+
+        std::uint32_t get_initial_num_localities() const override
+        {
+            return get_config().get_num_localities();
+        }
+
+        lcos::future<std::uint32_t> get_num_localities() const override
+        {
+            return agas_client_.get_num_localities_async();
+        }
+
+        std::uint32_t get_num_localities(hpx::launch::sync_policy,
+            components::component_type type, error_code& ec) const
+        {
+            return agas_client_.get_num_localities(type, ec);
+        }
+
+        lcos::future<std::uint32_t> get_num_localities(
+            components::component_type type) const
+        {
+            return agas_client_.get_num_localities_async(type);
+        }
```

Comment:
Moving all of those implementations into a .cpp file might reduce compile times and direct dependencies.